Treat nil values differently from missing keys

stuartc opened the pull request:
Hey! :wave: 

I have a proposed fix for #16, I've added a default to the `get/2` function which returns `:missing` when a key is missing instead of `nil` - this appears to disambiguate nils and missing keys.

I'm not _crazy_ about my use of an atom like `:missing` here, like you really never really know what a users data is going to be. Thoughts?

Fixes #16

stuartc committed:
Treat nil values differently from missing keys

corka149 commented:
You have a good solution. But indeed I am not so sure about using `:missing` as a `nil` alternative.

My suggestion would be: `:__jsonpatch_lib__missing_value__`. That should be unique enough (hopefully :D) .

corka149 reviewed (changes requested):
Change `:missing` to `:__jsonpatch_lib__missing_value__` or something else that is long and unique enough.

stuartc commented:
@corka149 sounds good to me, I've changed it to `:__jsonpatch_lib__missing_value__`.

Thanks for pointing out the `[nil]` scenario, I've added a test for that as well to get those changes exercised further.

corka149 commented:
Thanks again @stuartc ❤️  . I will create a release during the weekend.
